Objects/structseq.c

```diff
@@ -421,12 +420,18 @@ structseq_replace(PyStructSequence *self, PyObject *args, PyObject *kwargs)
             if (!key) {
                 goto error;
             }
-            PyObject *ob = _PyDict_Pop(kwargs, key, self->ob_item[i]);
-            Py_DECREF(key);
-            if (!ob) {
+            PyObject *ob;
+            if (PyDict_Pop(kwargs, key, &ob) < 0) {
+                Py_DECREF(key);
                 goto error;
             }
-            result->ob_item[i] = ob;
+            Py_DECREF(key);
+            if (ob != NULL) {
+                result->ob_item[i] = ob;
+            }
+            else {
+                result->ob_item[i] = Py_NewRef(self->ob_item[i]);
+            }
```

Comment:
I would write it in different way, but it is up to you.
```suggestion
            if (ob == NULL) {
                ob = Py_NewRef(self->ob_item[i]);
            }
            result->ob_item[i] = ob;
```